src/Symfony/Component/Security/Http/Firewall/LogoutListener.php

```diff
@@ -127,6 +131,18 @@ public function handle(GetResponseEvent $event)
      */
     protected function requiresLogout(Request $request)
     {
-        return $this->httpUtils->checkRequestPath($request, $this->options['logout_path']);
+        if (!isset($this->options['logout_path']) && null === $this->requestMatcher) {
+            return false;
+        }
```

Comment:
Should this condition be allowed at all? This seems invalid: you should pass one or the other. We could catch this with validation in `MainConfiguration` (and an exception here). We should probably also not allow *both* to be set.